oop_student_report_project.py

```diff
@@ -0,0 +1,180 @@
+'''
+Create a students report card class showing following detials as table
+- name
+- roll no.
+- Maths marks
+- physics marks
+- chemistry marks
+- eng marks
+- total percentage 
+
+Note: should not consider the class of the student in which
+they are studying but the table will be showing the a group of students of one class. 
+
+You can add more methods such as ID function that shows the ID card of a Student A with 
+RollNo. , Blood group, Address and Parents name, emailID.
+
+
+'''
+import time
+"""time is a python module which allows user to work with time by providing functionalities like getting the current time,
+ pausing the Program from executing, etc. """
+
+from prettytable import PrettyTable
+"""PrettyTable is a simple Python class library that alows user to easily display 
+   tabular data in a visually appealing ASCII table format.
+   Documentation link -----> https://pypi.org/project/prettytable/"""
+
+class Student:
+     """
+    Represents a student with various attributes such as name, roll number, marks in different subjects,
+    percentage, blood group, address, and parent information.
+
+    Attributes:
+        name (str): The student's full name.
+        roll_number (str): The student's roll number.
+        m_marks (int): Marks obtained in mathematics.
+        p_marks (int): Marks obtained in physics.
+        c_marks (int): Marks obtained in chemistry.
+        eng_marks (int): Marks obtained in English.
+        percent (float): Overall percentage calculated as the average of subject marks.
+        b_group (str): Blood group of the student.
+        address (str): Address of the student.
+        m_name (str): Mother's name.
+        f_name (str): Father's name.
+        email (str): Email address of the student.
+
+    Methods:
+        add_student(): Allows user input to set the attributes for a student instance.
+    """
+     def __init__(self):
+        self.name = None
+        self.roll_number = None
+        self.m_marks = None
```

Comment:
I can still see that you have not renamed all the variables here. m_marks means what?? Kindly rename them and give proper naming convention... 
```suggestion
        self.maths_marks = None
```

Reply:
Do it for all, don't rush to complete the code... take a closure look at each line then submit it.